e2e/test/E2EMsTestBase.cs

```diff
@@ -1,21 +1,28 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Diagnostics.Tracing;
 using Microsoft.ApplicationInsights.DataContracts;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 
+//Workers = 0 makes the test engine use one worker per available core. It does not mean to run serially.
+[assembly: Parallelize(Workers = 0, Scope = ExecutionScope.ClassLevel)]
+
```

Comment:
I read about ExecutionScope.ClassLevel - This means tests within a class do not run in parallel right? This would interfere with the existing Logger implementation. We would then have to create a logger per test and not per class. Why are we not doing ExecutionScope.MethodLevel?

Reply:
Yes, that is correct. The plan is to have tests run at `ExecutionScope.MethodLevel` eventually, but we had issues with logging and also saw some test failures. (test failures were mostly related to event hub listener that we have - for validating message flow).
I will ignore the logging failures for now, since the AI changes you are making will take a different approach to logging, and remove the need for us relying on console logs on vsts. So let me run it at method level and see if we have test failures.